File: src/main/scala-graphql/io/github/graphql/model/IssueCommentConnectionTO.scala
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import scala.collection.JavaConverters._

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-06T19:48:01+0800"
)
class IssueCommentConnectionTO(
    edges: Seq[IssueCommentEdgeTO],
    nodes: Seq[IssueCommentTO],
    @javax.validation.constraints.NotNull
    pageInfo: PageInfoTO,
    totalCount: Int
) {

    override def toString(): String = {
        Seq(
            if (edges != null) "edges: " + GraphQLRequestSerializer.getEntry(edges.asJava) else "",
            if (nodes != null) "nodes: " + GraphQLRequestSerializer.getEntry(nodes.asJava) else "",
            if (pageInfo != null) "pageInfo: " + GraphQLRequestSerializer.getEntry(pageInfo) else "",
            "totalCount: " + GraphQLRequestSerializer.getEntry(totalCount)
        ).filter(_ != "").mkString("{", ",", "}")
    }
}

object IssueCommentConnectionTO {

    def builder(): IssueCommentConnectionTO.Builder = new Builder()

    class Builder {

        private var edges: Seq[IssueCommentEdgeTO] = _
        private var nodes: Seq[IssueCommentTO] = _
        private var pageInfo: PageInfoTO = _
        private var totalCount: Int = _

        def setEdges(edges: Seq[IssueCommentEdgeTO]): Builder = {
            this.edges = edges
            this
        }

        def setNodes(nodes: Seq[IssueCommentTO]): Builder = {
            this.nodes = nodes
            this
        }

        def setPageInfo(pageInfo: PageInfoTO): Builder = {
            this.pageInfo = pageInfo
            this
        }

        def setTotalCount(totalCount: Int): Builder = {
            this.totalCount = totalCount
            this
        }

        def build(): IssueCommentConnectionTO = new IssueCommentConnectionTO(edges, nodes, pageInfo, totalCount)

    }
}

File: src/main/java-graphql/io/github/graphql/model/IssueCommentConnectionTO.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer;
import java.util.StringJoiner;

@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:47+0800"
)
public class IssueCommentConnectionTO implements java.io.Serializable {

    private java.util.List<IssueCommentEdgeTO> edges;
    private java.util.List<IssueCommentTO> nodes;
    @javax.validation.constraints.NotNull
    private PageInfoTO pageInfo;
    private int totalCount;

    public IssueCommentConnectionTO() {
    }

    public IssueCommentConnectionTO(java.util.List<IssueCommentEdgeTO> edges, java.util.List<IssueCommentTO> nodes, PageInfoTO pageInfo, int totalCount) {
        this.edges = edges;
        this.nodes = nodes;
        this.pageInfo = pageInfo;
        this.totalCount = totalCount;
    }

    public java.util.List<IssueCommentEdgeTO> getEdges() {
        return edges;
    }

    public java.util.List<IssueCommentTO> getNodes() {
        return nodes;
    }

    public PageInfoTO getPageInfo() {
        return pageInfo;
    }

    public int getTotalCount() {
        return totalCount;
    }


    @Override
    public String toString() {
        StringJoiner joiner = new StringJoiner(", ", "{ ", " }");
        if (edges != null) {
            joiner.add("edges: " + GraphQLRequestSerializer.getEntry(edges));
        }
        if (nodes != null) {
            joiner.add("nodes: " + GraphQLRequestSerializer.getEntry(nodes));
        }
        if (pageInfo != null) {
            joiner.add("pageInfo: " + GraphQLRequestSerializer.getEntry(pageInfo));
        }
        joiner.add("totalCount: " + GraphQLRequestSerializer.getEntry(totalCount));
        return joiner.toString();
    }

    public static IssueCommentConnectionTO.Builder builder() {
        return new IssueCommentConnectionTO.Builder();
    }

    public static class Builder {

        private java.util.List<IssueCommentEdgeTO> edges;
        private java.util.List<IssueCommentTO> nodes;
        private PageInfoTO pageInfo;
        private int totalCount;

        public Builder() {
        }

        public Builder setEdges(java.util.List<IssueCommentEdgeTO> edges) {
            this.edges = edges;
            return this;
        }

        public Builder setNodes(java.util.List<IssueCommentTO> nodes) {
            this.nodes = nodes;
            return this;
        }

        public Builder setPageInfo(PageInfoTO pageInfo) {
            this.pageInfo = pageInfo;
            return this;
        }

        public Builder setTotalCount(int totalCount) {
            this.totalCount = totalCount;
            return this;
        }


        public IssueCommentConnectionTO build() {
            return new IssueCommentConnectionTO(edges, nodes, pageInfo, totalCount);
        }

    }
}
